File: src/back-end-dotnet/HOB.Data/Entities/Product.cs
namespace HOB.Data.Entities;

public class Product
{
    public Guid ProductId { get; set; }
    public string SKU { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal UnitPrice { get; set; }
    public int StockQuantity { get; set; }
    public int LowStockThreshold { get; set; }
    public string? Category { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation property
    public virtual ICollection<Sale> Sales { get; set; } = new List<Sale>();
}

File: src/front-end-nextjs/hob-dashboard/lib/types/product.ts
export interface Product {
  productId: string;
  sku: string;
  name: string;
  description?: string;
  unitPrice: number;
  stockQuantity: number;
  lowStockThreshold: number;
  category?: string;
  isActive: boolean;
  isLowStock: boolean;
  createdAt: string;
  updatedAt: string;
  salesCount?: number;
}

export interface CreateProductRequest {
  sku: string;
  name: string;
  description?: string;
  unitPrice: number;
  stockQuantity: number;
  lowStockThreshold: number;
  category?: string;
}

export interface UpdateProductRequest {
  sku: string;
  name: string;
  description?: string;
  unitPrice: number;
  stockQuantity: number;
  lowStockThreshold: number;
  category?: string;
  isActive: boolean;
}
